File: 06/Java/TripleExpression/Subtract.java
package TripleExpression;

public class Subtract extends BinaryOperator implements TripleExpression {
    public Subtract(TripleExpression first, TripleExpression second) {
        super(first, second);
    }

    protected int apply(int a, int b) {
        return a - b;
    }
}

File: 08/CSharp/TripleExpressionApp/TripleExpressionApp/Subtract.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Subtract<T> : BinaryOperator<T>
    {
        public Subtract(ITripleExpression<T> first, ITripleExpression<T> second, IOperator<T> op)
        {
            fOp = first;
            sOp = second;
            base.op = op;
        }

        override protected T Apply(T a, T b)
        {
            return op.Subtract(a, b);
        }
    }
}
